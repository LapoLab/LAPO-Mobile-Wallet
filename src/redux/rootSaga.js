import { all } from 'redux-saga/effects';
import commonsSagas from '../commons/sagas';
import recoveryWordsSagas from '../modules/RecoveryWords/sagas';
import sendSagas from '../modules/Send/sagas';
import paperWordsConfirm from '../modules/PaperWordsConfirm/sagas';
import dashboard from '../modules/Dashboard/sagas';
import settings from '../modules/Settings/sagas';
import paperWords from '../modules/PaperWords/sagas';
import transactionsDetails from '../modules/TransactionDetails/sagas';
import chooseContact from '../modules/ChooseContact/sagas';
import splashScreen from '../modules/SplashScreen/sagas';
import switchContacts from '../modules/SwitchContacts/sagas';
import receive from '../modules/Receive/sagas';

// eslint-disable-next-line
const getListeners = (...args) => args.reduce((acc, nextArg) => [...acc, ...Object.values(nextArg).map(func => func())], []);

export default function* rootSaga() {
  yield all(
    getListeners(
      commonsSagas,
      recoveryWordsSagas,
      sendSagas,
      paperWordsConfirm,
      dashboard,
      settings,
      paperWords,
      transactionsDetails,
      chooseContact,
      splashScreen,
      switchContacts,
      receive,
    ),
  );
}
