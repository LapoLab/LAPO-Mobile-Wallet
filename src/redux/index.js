import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import storage from 'redux-persist/lib/storage';
import { persistStore, persistReducer } from 'redux-persist';
import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';
import rootReducer from './rootReducer';
import rootSagas from './rootSaga';

const persistConfig = {
  key: 'root',
  storage,
  stateReconciler: autoMergeLevel2,
  blacklist: [
    'modals',
    'dashboard',
    'paperWords',
    'chooseContact',
    'send',
    'transactionsDetails',
    'alert',
    'pinsRecovery',
  ],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const sagaMiddleware = createSagaMiddleware();
// eslint-disable-next-line no-undef
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const store = createStore(
  persistedReducer,
  composeEnhancers(applyMiddleware(sagaMiddleware)),
);
sagaMiddleware.run(rootSagas);

export const persistor = persistStore(store);
