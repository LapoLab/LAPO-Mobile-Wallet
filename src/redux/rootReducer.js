import { combineReducers } from 'redux';

import wallet from '../commons/reducers/wallet';
import pins from '../commons/reducers/pins';
import pinsRecovery from '../commons/reducers/pinsRecovery';
import modals from '../commons/reducers/modals';
import settings from '../commons/reducers/settings';
import notify from '../modules/Notify/reducers';
import paperConfirm from '../modules/PaperWordsConfirm/reducer';
import dashboard from '../modules/Dashboard/reducers';
import paperWords from '../modules/PaperWords/reducers';
import chooseContact from '../modules/ChooseContact/reducers';
import send from '../modules/Send/reducers';
import transactionsDetails from '../modules/TransactionDetails/reducer';
import alert from '../modules/Alert/reducers';

export default combineReducers({
  wallet,
  pins,
  pinsRecovery,
  notify,
  modals,
  settings,
  paperConfirm,
  dashboard,
  paperWords,
  chooseContact,
  send,
  transactionsDetails,
  alert,
});
