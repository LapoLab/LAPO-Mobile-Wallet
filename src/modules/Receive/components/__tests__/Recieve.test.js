// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../../../redux';
import Recieve from '../index';

describe('Recieve page', () => {
  test('renders snapshots', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Recieve />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
