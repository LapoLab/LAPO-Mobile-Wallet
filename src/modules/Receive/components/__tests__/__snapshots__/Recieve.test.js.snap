// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recieve page renders snapshots 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-end",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#EFF4FC",
        "borderTopLeftRadius": 12,
        "borderTopRightRadius": 12,
        "height": 450,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "#D8D8D8",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "space-between",
            "paddingRight": 15,
            "paddingVertical": 15,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "padding": 15,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/close.png",
              }
            }
            style={
              Object {
                "height": 15,
                "resizeMode": "contain",
                "width": 15,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#131313",
              "flex": 1,
              "fontFamily": "Lato-Bold",
              "fontSize": 18,
              "textAlign": "center",
            }
          }
        >
          Receive
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "padding": 15,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "borderTopColor": "#D8D8D8",
            "borderTopWidth": 1,
            "height": 110,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4183BD",
                "borderColor": "#BBCDEB",
                "borderRadius": 74,
                "borderWidth": 8,
                "height": 64,
                "justifyContent": "center",
                "marginHorizontal": 25,
                "opacity": 1,
                "width": "80%",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Lato-Regular",
                  "fontSize": 18,
                }
              }
            >
              SHARE IT!
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
